Molio/Source/Domain/UseCase/FetchAllPlaylistsUseCase/DefaultFetchAllPlaylistsUseCase.swift

```diff
@@ -0,0 +1,20 @@
+import Foundation
+
+final class DefaultFetchAllPlaylistsUseCase: FetchAllPlaylistsUseCase {
+    private let playlistRepository: any PlaylistRepository
+
+    init(
+        playlistRepository: any PlaylistRepository = DIContainer.shared.resolve()
+    ) {
+        self.playlistRepository = playlistRepository
+    }
+    
+    func execute() async -> [MolioPlaylist] {
+        return await withCheckedContinuation { continuation in
+            DispatchQueue.global().async {
+                let playlists = self.playlistRepository.fetchPlaylists() ?? []
+                continuation.resume(returning: playlists)
+            }
+        }
+    }
```

Comment:
```suggestion
    func execute() async -> [MolioPlaylist] {
        return await self.playlistRepository.fetchPlaylists() 
    }
```

맞습니다. 이 부분에서 사용되는 콜백 기반 비동기 코드가 없는 만큼.
 이렇게 쓰시면 어떨까요?

Reply:
비동기로 바꿀 가능성이 있기 때문에 이렇게 구현하였습니다. 현재 playlistService 프로토콜에도 async throws 적용될 것 같은데 그냥 돠두고 PlaylistService 작성할 때 바꾸는 건 어떨까요???????

Reply:
비동기 코드인 거랑 withCheckedContinuation를 사용하는 것이랑 무슨 상관인지 모르겠습니다. 혹시 withCheckedContinuation 어떨 때 사용하는지 잘 아시고 쓰시는 걸까요?